### PR TITLE
feat(types): add typescript definitions

### PR DESCRIPTION
**Pull request by elyukai**
I forked the TypeScript definitions for Nunjucks from https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/nunjucks and adjusted them for this Deno port to be able to use it with TypeScript more easily.

I hope this is alright, I happy to change anything that does not meet requirements.

Thank you!

### CODE REVIEW

File: mod.d.ts [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+      commentEnd?: string | undefined;
+    };
+  }
+
+  export function configure(options: ConfigureOptions): Environment;
+  export function configure(path: string | string[], options?: ConfigureOptions): Environment;
+
+  export function reset(): void;
+
+  export function compile: (message: string, lineno: number, colno: number, eagerCompile?: boolean) => Template;
```
Comment (valtlai) [inline mod.d.ts:173]: There’s two syntax errors on this line: should be `compile(` and `): Template`.

**Comment by valtlai** [comment]
Thank you a lot for this nice PR. Could you run `deno fmt` to autoformat the code? There’s a pair of syntax errors that need fixing first.

File: mod.d.ts (hunk 1 above)
Comment (elyukai) [reply mod.d.ts]: I’m really sorry, for some reason my editor didn’t show that – until I just reloaded the project. Should be fixed now.

**Comment by elyukai** [comment]
Happy to help! I did run the formatter and I also fixed other TypeScript-related issues that weren’t showing up before as well.

**Comment by valtlai** [comment]
Thanks! I’ll release a new version soon.

**Pull request merged**